guidance/models/transformers/_transformers.py

```diff
@@ -98,7 +100,20 @@ def __init__(
                     elif isinstance(token, str):
                         if hasattr(transformers_tokenizer, "convert_tokens_to_string"):
                             token_str = transformers_tokenizer.convert_tokens_to_string([token])
-                            byte_coded = token_str.encode()
+                            roundtrip_id = transformers_tokenizer.encode(token_str)[0]
+                            if roundtrip_id == i:
+                                byte_coded = token_str.encode()
+                            else:
+                                # TODO replace this with something more sensible
+                                if not fallback_decoder:
+                                    fallback_decoder = transformers_package.AutoTokenizer.from_pretrained(
+                                        "gpt2", use_fast=False
+                                    ).byte_decoder
+                                    
+                                byte_coded = bytes(
+                                    [fallback_decoder[c]
+                                     for c in transformers_tokenizer.convert_ids_to_tokens(i)]
+                                )
```

Comment:
Someone better at understanding tokenizers than me could help me understand why encoding unicode characters via .encode() doesn't work here, even though the characters are present in the vocab and .encode().decode() produces the same character.